tools/devops/automation/templates/packages/build.yml

```
@@ -0,0 +1,454 @@
+- task: PublishPipelineArtifact@1
+  displayName: 'Publish Build Artifacts'
+  inputs:
+    targetPath: $(Build.SourcesDirectory)/package
+    artifactName: package
+  continueOnError: true
+
+- task: PublishPipelineArtifact@1
+  displayName: 'Publish Build Internal Artifacts'
+  inputs:
+    targetPath: $(Build.SourcesDirectory)/package-internal 
+    artifactName: package-internal
+  continueOnError: true
+
+- bash: |
+    set -x
+    set -e
+
+    make -C $(Build.SourcesDirectory)/xamarin-macios/tests package-tests
+  displayName: 'Package Xamarin.mac tests'
+  condition: and(succeeded(), eq(variables['configuration.RunMacTests'], 'True'))
+  continueOnError: true # not a terrible blocking issue
+
+- task: PublishPipelineArtifact@1
+  displayName: 'Publish Xamarin.Mac tests'
+  inputs:
+    targetPath: $(Build.SourcesDirectory)/xamarin-macios/tests/*.7z
+    artifactName: package-internal
+  condition: and(succeeded(), eq(variables['configuration.RunMacTests'], 'True'))
+  continueOnError: true
+
+- bash: |
+    make -j8 -C $(Build.SourcesDirectory)/xamarin-macios/tools/apidiff jenkins-api-diff
+
+    # remove some files that do not need to be uploaded
+    cd $(Build.SourcesDirectory)/xamarin-macios/tools/apidiff/
+    rm -Rf *.exe *.pdb *.stamp *.zip *.sh ./references ./temp
+  displayName: 'API diff (from stable)'
+  condition: and(succeeded(), eq(variables['configuration.SkipPublicJenkins'], 'False'))
+  continueOnError: true
+  env:
+    BUILD_REVISION: 'jenkins'
+
+- task: ArchiveFiles@1
+  displayName: 'Archive API diff (from stable)'
+  inputs:
+    rootFolder: $(Build.SourcesDirectory)/xamarin-macios/tools/apidiff
+    includeRootFolder: false
+    archiveFile: '$(Build.ArtifactStagingDirectory)/apidiff-stable.zip'
+  condition: and(succeeded(), eq(variables['configuration.SkipPublicJenkins'], 'False'))
+  continueOnError: true
+
+- task: PublishPipelineArtifact@1
+  displayName: 'Publish API diff (from stable)'
+  inputs:
+    targetPath: '$(Build.ArtifactStagingDirectory)/apidiff-stable.zip'
+    artifactName: apidiff-stable
+  condition: and(succeeded(), eq(variables['configuration.SkipPublicJenkins'], 'False'))
+  continueOnError: true
+
+- bash: |
+    echo "Running tests on $AGENT_NAME"
+    echo "##vso[task.setvariable variable=TESTS_BOT;isOutput=true]$AGENT_NAME"
+
+    echo "##vso[task.setvariable variable=TESTS_RAN;isOutput=true]True"
+    rm -rf ~/.config/.mono/keypairs/
+
+    RC=0
+    make -C $(Build.SourcesDirectory)/xamarin-macios/tests "$TARGET" || RC=$?
+
+    if [ $RC -eq 0 ]; then
+      echo "##vso[task.setvariable variable=TESTS_JOBSTATUS;isOutput=true]Succeeded"
+    else
+      echo "##vso[task.setvariable variable=TESTS_JOBSTATUS;isOutput=true]Failed"
+    fi
+
+    if test -f "$(Build.SourcesDirectory)/xamarin-macios//jenkins/failure-stamp"; then
+      echo "Something went wrong:"
+      cat "$(Build.SourcesDirectory)/xamarin-macios//jenkins/pr-comments.md"
+      exit 1
+    fi
```

This will be green if the failure-stamp file wasn't created, you need to return RC:

```suggestion
    fi
    
    exit $RC
```

---

I've realised, we don't want that, that will fail the build, but we want to continue to be able to report the results, else we will get out.